scripts/migrate_registration_logs.py

```diff
@@ -1,36 +1,75 @@
+"""
+This migration will change params['node'] to be the node (if it isn't node already), and will set
+params['registration'] equal to the associated registration.
+"""
+
 import logging
 
 from modularodm import Q
 
-from website.models import NodeLog
+from website.models import NodeLog, Node, RegistrationApproval
 from website.app import init_app
 
 from scripts import utils as script_utils
+from framework.mongo import database as db
 from framework.transactions.context import TokuTransaction
 
 logger = logging.getLogger(__name__)
 logging.basicConfig(level=logging.INFO)
 
 
 def get_targets():
+    """
+    Fetches all registration-related logs except for project_registered.
+
+    project_registered log is not included because params already correct.
+    """
     logs = NodeLog.find(
-            Q('action', 'eq', 'embargo_approved') |
-            Q('action', 'eq', 'embargo_initiated') |
-            Q('action', 'eq', 'retraction_approved') |
-            Q('action', 'eq', 'retraction_initiated') |
-            Q('action', 'eq', 'registration_initiated')
+        Q('action', 'eq', 'registration_initiated') |
+        Q('action', 'eq', 'registration_approved') |
+        Q('action', 'eq', 'registration_cancelled') |  # On staging, there are a few inconsistencies with these.  Majority of params['node'] are registrations, but a handful are nodes.
+        Q('action', 'eq', 'retraction_initiated') |
+        Q('action', 'eq', 'retraction_approved') |  # params['node'] is already equal to node.  Adds registration_field below.  Will be slow.
+        Q('action', 'eq', 'retraction_cancelled') |
+        Q('action', 'eq', 'embargo_initiated') |
+        Q('action', 'eq', 'embargo_approved') |
+        Q('action', 'eq', 'embargo_completed') |
+        Q('action', 'eq', 'embargo_cancelled')
     )
     return logs
 
 
+def get_registered_from(registration):
+    """
+    Gets node registration was registered from.  Handles deleted registrations where registered_from is null.
+
+    """
+    if registration.registered_from:
+        return registration.registered_from_id
+    else:
+        first_log = db['node'].find_one({'_id': registration._id})['logs'][0]
+        return NodeLog.load(first_log).params['node']
```

Comment:
Based on 638f73167fb, there was a time when we didn't add the `node` param for top-level projects. So you can use `params['project']` in those cases.